Lib/traceback.py

```diff
@@ -698,6 +698,8 @@ def _should_show_carets(self, start_offset, end_offset, all_lines, anchors):
         with suppress(SyntaxError, ImportError):
             import ast
             tree = ast.parse('\n'.join(all_lines))
+            if not tree.body:
```

Comment:
Do you think that we should be extra safe and check also the type to avoid an error on getting the `body` attribute?

```suggestion
            if not isinstance(tree, ast.Module) or not tree.body:
```

Reply:
Sorry I forgot to reply: AFAIR, no, because ast.parse always returns a module object when used in "exec" mode. You may have other kind of ast nodes but only if you change the parsing mode.

Reply:
Ok, so `isinstance(tree, ast.Module)` test is redundant. Thanks for checking.